.gitattributes

```diff
@@ -1,2 +1,5 @@
+# Forces all files which git considers text files to use LF line endings
+* text=auto eol=lf
```

Comment:
this won't make git convert them, will it?

Reply:
I think it enables end-of-line conversion, if I read the documentation right. Normally you can edit it with the `autocrlf` git config.

Reply:
If you have concerns, you can also look at how other projects handle this, e.g. godot:
https://github.com/godotengine/godot/blob/master/.gitattributes

Reply:
If I understand correctly this means people will get files with CRLF when they clone MT on Windows.
Honestly not sure if we want this?

Reply:
> If I understand correctly this means people will get files with CRLF when they clone MT on Windows.

I thought it's opposite, LF for Windows users, since `eol=lf`.
The documentation for `eol` says:

> Set to string value "lf"
>                This setting uses the same line endings in the working directory as in the index when the file is checked out

Only if `eol` is unspecified and `text=auto`, it would convert all text files to CRLF for Windows users by default.
It's the same as if you would put the `text` attribute to a single file type and not specify `eol`.
So `* text=auto eol=lf` shouldn't change much for Windows users. (Though I don't know whether Windows on itself somehow converts files to CRLF, maybe someone who uses windows should test this.)

> Honestly not sure if we want this?

Ok, let's wait for other reviews, maybe someone wants to keep CRLF files, then the line ending can be specified for every filetype manually. e.g. `*.cpp text eol=lf` instead of `text=auto`.

Reply:
FWIW I presently see no reason to keep CRLF files.

Relevant manpage excerpts:

##### `text`

           Set to string value "auto"
               When text is set to "auto", the path is marked for automatic
               end-of-line conversion. If Git decides that the content is
               text, its line endings are converted to LF on checkin. When the
               file has been committed with CRLF, no conversion is done.

##### `eol`

           Set to string value "lf"
               This setting forces Git to normalize line endings to LF on
               checkin and prevents conversion to CRLF when the file is
               checked out.

So this should be fine.

Reply:
> prevents conversion to CRLF when the file is checked out.

This is what I wanted to know.